### ISSUE TITLE
Buggy when starting from a fullscreen window

### ISSUE THREAD

**Issue by Dlurak** [hyprwm/hyprlock#332]
# Steps to reproduce

1. Put a Hyprland window into fullscreen either via `hyprctl dispatch fullscreen 0` or using a keyboard shortcut
2. Activate Hyprlock

# Bugs

## Background Picture and Labels Not Applied in Fullscreen

- **Behavior**: Hyprlock successfully locks. I can type my password, press enter, and it unlocks. However, the configured background picture and other labels are not applied when the window is in fullscreen. Once the window is no longer in fullscreen, Hyprlock functions correctly.
- **Additional Issue**: Even though the screen is locked, I can still de-maximize the window using my keyboard shortcut.

## Workspace Switching and Input Issues

- **Behavior*:*
	1. If the maximized window is on workspace 1 and the screen is locked, I can switch to another workspace (e.g., workspace 2) using a keyboard shortcut. On workspace 2, the background picture and other labels are displayed correctly.
	2. However, I can switch back to workspace 1 and see the maximized window again.
	3. After switching workspaces for the first time, I am unable to input my password anymore.

**Comment by vaxerski** [comment]
what's your hyprland ver

**Comment by Dlurak** [comment]
```bash
$ hyprctl version
Hyprland, built from branch  at commit cba1ade848feac44b2eda677503900639581c3f4  (props: bump version to 0.40.0).
Date: Sat May 4 15:42:32 2024
Tag: v0.40.0, commits: 4606

flags: (if any)
```

**Comment by bullsbarry** [comment]
I also am seeing this on the same hyprland version. It also happens if using swaylock.

**Comment by Dlanis** [comment]
I'm having same bug. For now I using this keybind:
`if [[ "$(hyprctl -j activewindow | jq -r '.fullscreen')" != "true" ]] ; then hyprlock ; else hyprctl notify 0 5000 0 " Locking in fullscreen is bugged" ; fi`

**Comment by dr0bz** [comment]
I can even press my hotkey for fullscreen, when hyprlock is running and the window goes fullscreen above the hyprlock.

**Comment by DiedByDisgust** [comment]
> I can even press my hotkey for fullscreen, when hyprlock is running and the window goes fullscreen above the hyprlock. Works also in sway.
> 
> 

I can confirm. Also any other Hyprland bind works while hyprlocked.

**Comment by Dlanis** [comment]
> > I can even press my hotkey for fullscreen, when hyprlock is running and the window goes fullscreen above the hyprlock. Works also in sway.
> 
> I can confirm. Also any other Hyprland's bind works while hyprlocked.

This bug has already been patched. https://github.com/hyprwm/Hyprland/commit/cddeec47a1fc0e70d8598fd10c29cd0e9489999f

**Comment by dr0bz** [comment]
> This bug has already been patched. https://github.com/hyprwm/Hyprland/commit/cddeec47a1fc0e70d8598fd10c29cd0e9489999f

latest main issue still exists.

**Comment by Dlanis** [comment]
> > This bug has already been patched. [hyprwm/Hyprland@cddeec4](https://github.com/hyprwm/Hyprland/commit/cddeec47a1fc0e70d8598fd10c29cd0e9489999f)
> 
> current main ([c5b8ad0](https://github.com/hyprwm/hyprlock/commit/c5b8ad03d03ddbd2b0ff8615c2f6dba31374b6a8)) issue still exists. I've started hyprlock from cmd.

Yes main still here, I only was talking about bug with keybinds.

**Comment by dr0bz** [comment]
>Yes main one still here, I only was talking about bug with keybinds.

i'm also talking about the keybindings bug:
- i'm starting the hyprlock
- pressing a keybinding for `fullscreen`
- a window comes above the hyprlock (back and forth while pressing the keybinding)

**Comment by Dlanis** [comment]
> > Yes main one still here, I only was talking about bug with keybinds.
> 
> i'm also talking about the keybindings bug:
> 
>     * i'm starting the hyprlock
> 
>     * pressing a keybinding for `fullscreen`
> 
>     * a window comes above the hyprlock (back and forth while pressing the keybinding)

I compiled Hyprland v0.41.0 with the patch and keybinds bug fixed for me. The patch has been merged in latest version [0.41.1](https://github.com/hyprwm/Hyprland/releases/tag/v0.41.1).

**Comment by dr0bz** [comment]
@Dlanis OK, misunderstanding here. I only tested while compiling the hyprlock, not Hyprland.

**Comment by alba4k** [comment]
Works fine for me, is this fixed?

**Comment by Dlurak** [comment]
For me too
